ui/src/views/Documentation/Changelog/index.jsx

```diff
@@ -0,0 +1,227 @@
+};
+
+const FILTERS = ['version', 'from', 'to', 'q', 'all', 'audience'];
+
+@withRouter
+@withStyles(theme => ({
+  md: {
+    '& strong': {
+      background: theme.palette.secondary.main,
+      fontWeight: 'bold',
+    },
+  },
+  cbx: {
+    marginTop: theme.spacing(2),
+  },
+}))
+export default class Changelog extends Component {
+  static propTypes = {};
+
+  constructor(props) {
+    super(props);
+
+    this.sections = parseMarkdownIntoSections(ChangelogMd);
+    this.versions = [
+      ...new Set([...this.sections.map(section => section.version)]),
+    ].map(version => ({ label: version, value: version }));
+
+    this.audiences = [
+      ...new Set([...this.sections.map(section => section.audience)]),
+    ].map(audience => ({ label: titleCase(audience), value: audience }));
+
+    const search = new URLSearchParams(this.props.location.search);
+
+    this.state = {
+      version: search.get('version', ''),
+      from: search.get('from', ''),
+      to: search.get('to', ''),
+      q: search.get('q', ''),
+      all: search.get('all', ''),
+      audience: search.get('audience', ''),
+    };
+  }
+
+  filteredSections = memoize(
+    ({ version, from, to, q, all, audience }) => {
+      return this.sections.filter(section => {
+        if (audience && section.audience !== audience) {
+          return false;
+        }
+
+        if (q && !section.html.includes(q)) {
+          return false;
+        }
+
+        if (version) {
+          return section.version.includes(version);
+        }
+
+        if (!all || from || to) {
+          if (!validate(section.version)) {
+            return false;
+          }
+
+          const releaseFrom = from || (all ? '0.0.0' : '45.0.0');
+          const releaseTo = to || '999.999.999';
+
+          if (
+            compareVersions(section.version, releaseFrom) < 0 ||
+            compareVersions(section.version, releaseTo) > 0
+          ) {
+            return false;
+          }
+        }
+
+        return true;
+      });
+    },
+    {
+      serializer: ({ version, from, to, q, all, audience }) =>
+        `${version}-${from}-${to}-${q}-${all ? '1' : ''}-${audience}`,
+    }
+  );
+
+  componentDidUpdate(prevProps, prevState) {
+    if (FILTERS.map(key => prevState[key] !== this.state[key]).some(a => a)) {
+      this.updateUrl();
+    }
+  }
+
+  updateUrl() {
+    const { history, location } = this.props;
+    const newSearchParams = new URLSearchParams(location.search);
+
+    FILTERS.forEach(key => {
+      if (this.state[key]) {
+        newSearchParams.set(key, this.state[key]);
+      } else {
+        newSearchParams.delete(key);
+      }
+    });
+
+    history.push({ search: newSearchParams.toString() });
+  }
+
+  render() {
+    const dropdowns = {
+      version: 'Choose version',
+      from: 'Version from',
+      to: 'Version to',
+    };
+    const maybeHighlight = (text, q) =>
+      q ? text.replace(new RegExp(`(${q})(?![^<>]*>)`, 'gi'), '**$1**') : text;
```

Comment:
this was a hack, as `<mark>$1</mark>` would have required to enable html tags in markdown parser. <strong> works pretty much the same with a bit of styling, but not so semantic..